c++/src/TypeImpl.cc

```diff
@@ -218,7 +227,19 @@ namespace orc {
         if (i != 0) {
           result += ",";
         }
-        result += fieldNames[i];
+        if (isUnquotedFieldName(fieldNames[i])) {
+          result += fieldNames[i];
+        } else {
+          std::string name(fieldNames[i]);
+          size_t pos = 0;
+          while ((pos = name.find("`", pos)) != std::string::npos) {
+            name.replace(pos, 1, "``");
```

Comment:
If we call toString() to a type with quote and get a string. Can we call buildTypeFromString() on that string to get back the original type object?

Reply:
Sorry for the late reply. I got the answer from your test case (TEST(TestType, quotedFieldNames)) already.